File: examples/Delphi/Compound/ModelEvolution/Common/dSystem.pas
unit dSystem;

interface

uses
  Classes,
  Controls,
  Forms,
  BoldHandle,
  BoldPersistenceHandle,
  BoldPersistenceHandleDB,
  BoldModel,
  BoldHandles,
  BoldSubscription,
  BoldSystemHandle,
  BoldUMLModelLink,
  BoldUMLRose98Link,
  BoldActions,
  ActnList,
  BoldHandleAction, BoldAbstractObjectUpgraderHandle,
  BoldObjectUpgraderHandle, BoldDBActions, BoldIBDatabaseAction;

type
  TdmSystem = class(TDataModule)
    SystemHandle: TBoldSystemHandle;
    ActionList1: TActionList;
    BoldUpdateDBAction1: TBoldUpdateDBAction;
    BoldActivateSystemAction1: TBoldActivateSystemAction;
    BoldIBDatabaseAction1: TBoldIBDatabaseAction;
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  dmSystem: TdmSystem;

implementation

uses dSystemTypeInfo, dPersistence;

{$R *.DFM}

end.

File: examples/CBuilder/Compound/ModelEvolution/Common/dSystem.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "dSystem.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "BoldHandles"
#pragma link "BoldSubscription"
#pragma link "BoldSystemHandle"
#pragma link "BoldActions"
#pragma link "BoldDBActions"
#pragma link "BoldHandleAction"
#pragma resource "*.dfm"
TdmSystem *dmSystem;
//---------------------------------------------------------------------------
__fastcall TdmSystem::TdmSystem(TComponent* Owner)
  : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
